app/values/csv_report.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+# Convert an enumerable of resources into a CSV by converting it to a hash,
+# merging in imported properties, and selecting each property as a field. Right
+# now useful for ad-hoc reports, but could be used as a tool for generic report
+# generation.
+class CSVReport
+  attr_reader :resources, :fields
+  # @param resources [Enumerable<Valkyrie::Resource>] resources to convert to a
+  #   CSV
+  # @param fields [Array<Symbol>] fields which act as the header of the CSV
+  def initialize(resources, fields: [:title])
+    @resources = resources
+    @fields = fields
+  end
+
+  def csv_rows
+    @csv_rows ||= resources.lazy.map do |resource|
+      Row.new(resource)
+    end
+  end
+
+  def write(path:)
+    CSV.open(path, "w") do |csv|
+      csv << headers
+      csv_rows.each do |row|
+        csv << row.to_h.values_at(*fields)
+      end
+    end
+  end
+
+  def headers
+    fields.map(&:to_s).map(&:humanize)
+  end
+
+  class Row
+    attr_reader :resource
+    def initialize(resource)
+      @resource = resource
+    end
+
+    def to_h
+      Hash[
+        # __attributes__ is much faster and doesn't include nil values.
+        resource.__attributes__.merge(attributes_without_reserved(imported_metadata)).map do |key, values|
+          values = Array.wrap(values).map(&:to_s).join(", ")
```

Comment:
Is this separating values with a comma? Can we do that with a CSV? Maybe add a field to the test with multiple values.